### ISSUE TITLE
Syntax errors are not reported by MEvaluate on Windows

### ISSUE THREAD

**Issue by szhorvat** [rsmenon/MATLink#54]
Syntax errors are not reported by MEvaluate on Windows.

For example,

```
MEvaluate["1+"]
MEvaluate["eig(1"]
```

both return `Null` with no messages issued.

Non-syntax errors are reported correctly (e.g. `MEvaluate["eig"]`).

**Comment by szhorvat** [comment]
This happens because `engEvaluate[]` does not return any outputs that contain backslashes.  Errors reference files and the path separator is `\`, so error are not shown.

**Comment by szhorvat** [comment]
This is in fact a problem on all platforms, not just Windows.  Example evaluated on Mac:

```
In[9]:= MEvaluate["s='\\asd'"]

Out[9]= ">> 
s =

\.07sd

"

In[10]:= MGet["s"]

Out[10]= "\\asd"
```

It happens because `MLPutString()` tries to interpret backslash escape sequences, [as documented](http://reference.wolfram.com/mathematica/ref/c/MLPutString.html).  Note that `MLPutUTF8String()` does not interpret escape sequences.

It is only on Windows that it affects error printing though.

**Comment by szhorvat** [comment]
Fixed by 8bdf6362cd7c1f7ad3d4dfddbeea14da70546ddc
